packages/plugin-core/src/commands/CopyAsCommand.ts

```diff
@@ -0,0 +1,78 @@
+import { assertUnreachable } from "@dendronhq/common-all";
+import {
+  CopyAsFormat,
+  getAllCopyAsFormat,
+  JSONV2PodConfig,
+  MarkdownV2PodConfig,
+  PodExportScope,
+  PodV2Types,
+} from "@dendronhq/pods-core";
+import _ from "lodash";
+import { PodCommandFactory } from "../components/pods/PodCommandFactory";
+import { PodUIControls } from "../components/pods/PodControls";
+import { DENDRON_COMMANDS } from "../constants";
+import { VSCodeUtils } from "../vsCodeUtils";
+import { BasicCommand, CodeCommandInstance } from "./base";
+
+type CommandOutput = void;
+type CommandOpts = CodeCommandInstance;
+
+/**
+ * Command that will find the appropriate export command to run, and then run
+ * it. This is the UI entry point for all export pod functionality.
+ */
+export class CopyAsCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput,
+  CopyAsFormat
+> {
+  public format: CopyAsFormat[];
+  key = DENDRON_COMMANDS.COPY_AS.key;
+
+  constructor(_name?: string) {
+    super(_name);
+    this.format = getAllCopyAsFormat();
+  }
+
+  async sanityCheck() {
+    if (_.isUndefined(VSCodeUtils.getActiveTextEditor())) {
+      return "you must have a note open to execute this command";
+    }
+    return;
+  }
+
+  async gatherInputs(copyAsFormat?: CopyAsFormat) {
+    const format =
+      copyAsFormat || (await PodUIControls.promptToSelectCopyAsFormat());
+
+    if (!format) {
+      return;
+    }
+    switch (format) {
+      case CopyAsFormat.JSON: {
+        const config: JSONV2PodConfig = {
+          destination: "clipboard",
+          exportScope: PodExportScope.Note,
+          podType: PodV2Types.JSONExportV2,
+          podId: "copyAs.json", // dummy value, required property
+        };
+        return PodCommandFactory.createPodCommandForStoredConfig({ config });
+      }
+      case CopyAsFormat.MARKDOWN: {
+        const config: MarkdownV2PodConfig = {
+          destination: "clipboard",
+          exportScope: PodExportScope.Note,
+          podType: PodV2Types.MarkdownExportV2,
+          podId: "copyAs.markdown", // dummy value, required property
+        };
+        return PodCommandFactory.createPodCommandForStoredConfig({ config });
+      }
+      default:
+        assertUnreachable(format);
```

Comment:
When defining a keybinding and mispelling the format args the user can reach this, but the error message isn't informing.

Can we instead pop an error message saying "X" is not a valid copy as format and maybe suggest to check keybindings.json to make sure there's no typo? since that's the only way we can reach this path I _think_